### ISSUE TITLE
Evaluation Metrics

### ISSUE THREAD

**Issue by geringer** [awslabs/autogluon#205]
Hey guys,

I am trying to perform a tabular classification on a [kaggle dataset](https://www.kaggle.com/c/costa-rican-household-poverty-prediction) that I would like to use the f1 score for. when I pass 

```
eval_metric = 'f1'
```

 I am met with an error like

```
ValueError: Target is multiclass but average='binary'. Please choose another average setting, one of [None, 'micro', 'macro', 'weighted'].
```

but average is a parameter of sklearn.metrics.f1_score, so when I tried 

```
from sklearn.metrics import f1_score
...
eval_metric = f1_score(average = 'macro') #or None
```

I am required to pass the y values, which I don't have yet. 

Please advise!

Here is my full code

```
from autogluon import TabularPrediction as task
from sklearn.metrics import f1_score
train_data = task.Dataset(file_path="~/Datasets/costa-rican-household-poverty-prediction/train.csv") 

label_column = 'Target'
savedir = './ag_models/cr_poverty/' 

nn_options = {
    'num_epochs': 10, 
    'learning_rate': ag.space.Real(1e-4, 1e-2, default=5e-4, log=True),
    'activation': ag.space.Categorical('relu', 'softrelu', 'tanh'), 
    'layers': ag.space.Categorical([100],[1000],[200,100],[300,200,100]),
    'dropout_prob': ag.space.Real(0.0, 0.5, default=0.1), 
}

hyperparameters = {'NN': nn_options} 

time_limits = 20*60 
num_trials = 20 
search_strategy = 'skopt'


predictor = task.fit(train_data=train_data, label=label_column,
                     output_directory=savedir, time_limits=time_limits, num_trials=num_trials,
                     hyperparameter_tune=True, hyperparameters=hyperparameters,
                     search_strategy=search_strategy, eval_metric = 'f1')

results = predictor.fit_summary()

**Comment by Innixma** [comment]
Can you try:

`eval_metric = 'f1_macro'`

This should work. The documentation has to be updated to include this as an option.

**Comment by Innixma** [comment]
Documentation improvements to make this clearer have been merged and are available in the new version 0.0.4.

**Comment by Innixma** [comment]
Marking this issue as resolved. Please re-open if you still are experiencing problems in 0.0.4.

Best,
Nick
